deployment/ccip/changeset/cs_chain_contracts.go

```diff
@@ -4,35 +4,175 @@ import (
 	"bytes"
 	"context"
 	"encoding/hex"
+	"errors"
 	"fmt"
 	"math/big"
 
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
 	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/mcms"
 	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
 
 	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
 	"github.com/smartcontractkit/chainlink/deployment"
 	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset/internal"
 	commoncs "github.com/smartcontractkit/chainlink/deployment/common/changeset"
 	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/ccipevm"
 	cctypes "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/types"
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/fee_quoter"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/nonce_manager"
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/offramp"
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/onramp"
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/router"
 )
 
 var (
-	_ deployment.ChangeSet[UpdateOnRampDestsConfig]    = UpdateOnRampsDests
-	_ deployment.ChangeSet[UpdateOffRampSourcesConfig] = UpdateOffRampSources
-	_ deployment.ChangeSet[UpdateRouterRampsConfig]    = UpdateRouterRamps
-	_ deployment.ChangeSet[UpdateFeeQuoterDestsConfig] = UpdateFeeQuoterDests
-	_ deployment.ChangeSet[SetOCR3OffRampConfig]       = SetOCR3OffRamp
+	_ deployment.ChangeSet[UpdateOnRampDestsConfig]     = UpdateOnRampsDests
+	_ deployment.ChangeSet[UpdateOffRampSourcesConfig]  = UpdateOffRampSources
+	_ deployment.ChangeSet[UpdateRouterRampsConfig]     = UpdateRouterRamps
+	_ deployment.ChangeSet[UpdateFeeQuoterDestsConfig]  = UpdateFeeQuoterDests
+	_ deployment.ChangeSet[SetOCR3OffRampConfig]        = SetOCR3OffRamp
+	_ deployment.ChangeSet[UpdateFeeQuoterPricesConfig] = UpdateFeeQuoterPricesCS
+	_ deployment.ChangeSet[UpdateNonceManagerConfig]    = UpdateNonceManagersCS
 )
 
+type UpdateNonceManagerConfig struct {
+	UpdatesByChain map[uint64]NonceManagerUpdate // source -> dest -> update
+	MCMS           *MCMSConfig
+}
+
+type NonceManagerUpdate struct {
+	AddedAuthCallers   []common.Address
+	RemovedAuthCallers []common.Address
+	PreviousRampsArgs  []nonce_manager.NonceManagerPreviousRampsArgs
```

Comment:
I think in some cases we can we can look up the relevant addresses instead of passing them to make it more robust. Here I think it'll be simpler if we expose just a boolean flag like EnablePrevRamps (and perhaps a DisablePrevRamps flag which would set them to zero) which would look up the relevant prev ramps on the chain (and fail if there are no such legacy ramps). Similar to the approach I took for the TestRouter bool, that way you can't mess it up